### DISCUSSION TITLE
Issues with uzf observations...

### DISCUSSION THREAD

**Discussion by porterma** [modflowpy/flopy#1463] [General]
I am seeing some odd behavior when I try to set up observations in the uzf package using flopy 3.3.5.

Here is my code for the uzf package:

```
name = 'mf'
uzf_obs_input = f'{name}_uzf.obs'
flopy.mf6.ModflowGwfuzf(
    gwf,
    print_flows=True,
    save_flows=True,
    wc_filerecord=f'{name}_uzf.wc',
    budget_filerecord=f'{name}_uzf.cbb',
    observations=uzf_obs_input,
    nuzfcells=nuzfcells,
    ntrailwaves=ntrailwaves,
    nwavesets=nwavesets,
    packagedata=uzf_packagedata,
)
```

When I run `sim.write_simulation()` the mf.uzf package is created with the following options:

```
BEGIN options
  PRINT_FLOWS
  SAVE_FLOWS
  WATER_CONTENT  FILEOUT  mf_uzf.wc
  BUDGET  FILEOUT  mf_uzf.cbb
  OBS6  FILEIN  mf.uzf.obs
END options
```

Note that the OBS6 FILEIN is mf.uzf.obs not mf_uzf.obs as specified in the code. 

Additionally, flopy creates a mf.uzf.obs file that looks like this:

```
# File generated by Flopy version 3.3.5 on 07/22/2022 at 15:13:24.
BEGIN options
END options

BEGIN continuous  FILEOUT  m
  mf_uzf.obs
END continuous  FILEOUT  m
```

The expected behavior for the code block above is to produce a mf.uzf file with the following options:

```
BEGIN options
  PRINT_FLOWS
  SAVE_FLOWS
  WATER_CONTENT  FILEOUT  mf_uzf.wc
  BUDGET  FILEOUT  mf_uzf.cbb
  OBS6  FILEIN  mf_uzf.obs
END options
```

So that the modflow simulation reads my previously made mf_uzf.obs file that looks like this:

```
BEGIN OPTIONS
END OPTIONS

BEGIN CONTINUOUS FILEOUT  mf_uzf_gwrch.csv
uzf-gwrch 5
uzf-gwrch 6
uzf-gwrch 7
uzf-gwrch 8
uzf-gwrch 9
uzf-gwrch 10
.
.
.
uzf-gwrch 1641
uzf-gwrch 1650
uzf-gwrch 1651
uzf-gwrch 1652
uzf-gwrch 1653
uzf-gwrch 1660
END CONTINUOUS
```

I tried tracking the issue down in mfgwfuzf.py, but didn't make much progress.

Thoughts?

Thanks!

**Comment by christianlangevin** [comment]
Hey @porterma, have you seen this notebook?  It should give you the syntax you want.  https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_mf6_obs_ts_tas.ipynb

Information is also here:  https://flopy.readthedocs.io/en/3.3.5/_notebooks/tutorial02_mf6_data.html

**Comment by porterma** [comment]
I thought I looked through that example, but apparently not close enough. Thanks for pointing me in the right direction.

So it appears the observations argument is looking for a dictionary to create the observation input file. In my case where I already created the observation input file, is there a way to have flopy just point to it?

I feel like in the past I was able to just point to an observation input file when creating a package in flopy.

Thanks!

**Reply by christianlangevin** [reply]
@spaulins-usgs, is this an option?

**Reply by scottrp** [reply]
@langevin-usgs no, flopy lets you point to preexisting external data, but currently does not support pointing to preexisting packages.  Though I think such a feature would be fairly straightforward to add.
